Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp

```diff
@@ -70,24 +70,29 @@ static inline void fillRTCCodecStats(RTCStatsReport::CodecStats& stats, const Gs
         stats.payloadType = value;
     if (gst_structure_get_uint(structure, "clock-rate", &value))
         stats.clockRate = value;
+    if (gst_structure_get_uint(structure, "channels", &value))
+        stats.channels = value;
+
+    if (const char* sdpFmtpLine = gst_structure_get_string(structure, "sdp-fmtp-line"))
+        stats.sdpFmtpLine = String::fromLatin1(sdpFmtpLine);
+
+    if (const char* mimeType = gst_structure_get_string(structure, "mime-type"))
+        stats.mimeType = String::fromLatin1(mimeType);
 
     // FIXME:
-    // stats.mimeType =
-    // stats.channels =
-    // stats.sdpFmtpLine =
     // stats.implementation =
 }
 
-static inline void fillRemoteInboundRTPStreamStats(RTCStatsReport::RemoteInboundRtpStreamStats& stats, const GstStructure* structure, const GstStructure* additionalStats)
+static inline void fillReceivedRTPStreamStats(RTCStatsReport::ReceivedRtpStreamStats& stats, const GstStructure* structure)
 {
-    UNUSED_PARAM(additionalStats);
-    unsigned value;
-    if (gst_structure_get_uint(structure, "ssrc", &value))
-        stats.ssrc = value;
+    fillRTCRTPStreamStats(stats, structure);
 
-    double jitter;
-    if (gst_structure_get_double(structure, "jitter", &jitter))
-        stats.jitter = jitter;
+    GstStructure* rtpSourceStats;
+    gst_structure_get(structure, "gst-rtpsource-stats", GST_TYPE_STRUCTURE, &rtpSourceStats, nullptr);
```

Comment:
Can this structure be null? If it can, then protect the access below.

Reply:
no it can't